test/functional/lua/help_spec.lua

```diff
@@ -0,0 +1,54 @@
+-- Tests for gen_help_html.lua. Validates :help tags/links and HTML doc generation.
+--
+-- TODO: extract parts of gen_help_html.lua into Nvim stdlib?
+
+local helpers = require('test.functional.helpers')(after_each)
+local clear = helpers.clear
+local exec_lua = helpers.exec_lua
+local eq = helpers.eq
+local ok = helpers.ok
+
+describe(':help docs', function()
+  before_each(clear)
+  it('validate', function()
+    -- If this test fails, try these steps (in order):
+    -- 1. Try to fix/cleanup the :help docs, especially Nvim-owned :help docs.
+    -- 2. Try to fix the parser: https://github.com/vigoux/tree-sitter-vimdoc
+    -- 3. File a parser bug, and adjust the tolerance of this test in the meantime.
+
+    local rv = exec_lua([[return require('scripts.gen_help_html').validate('./build/runtime/doc')]])
```

Comment:
Test coverage to (1) exercise `gen_help_html.lua` and (2) validate `|tag|` links in :help docs.

In the future it probably makes sense to move parts of `gen_help_html.lua` into the stdlib.